docs/hooks-new-language.md

```diff
@@ -17,57 +17,59 @@ Dredd hooks language abstraction bridge comes with [the language agnostic test s
+    - after transaction
+    - after each transaction
+    - after all transactions
 - When CLI command is executed
-  - it loads files specified as CLI server arguments
-    - it exposes API similar to those in [Ruby](hooks-ruby.md), [Python](hooks-python.md) and [Node.js](hooks-nodejs.md) to each loaded file
-    - it registers functions declared in files for later execution
-
-  - starts a TCP socket server and starts listening on `http://127.0.0.1:61321`.
-
+    - it loads files specified as CLI server arguments
+        - it exposes API similar to those in [Ruby](hooks-ruby.md), [Python](hooks-python.md) and [Node.js](hooks-nodejs.md) to each loaded file
+        - it registers functions declared in files for later execution
+    - starts a TCP socket server and starts listening on `http://127.0.0.1:61321`.
 - When any data is received by the server
-  - Adds every received character to a buffer
+    - Adds every received character to a buffer
+    - When delimiting newline (`\n`) character is received
+        - It parses the [message](#tcp-socket-message-format) in the buffer as JSON
+        - It looks for `event` key in received object and executes appropriate registered hooks functions
+    - When the hook function is being executed
+        - It passes value of `data` key from received object to the executed function
+        - Hook function is able to modify data
+    - When function was executed
+        - It should serialize message to JSON
+        - Send the serialized message back to the socket with same `uuid` as received
+        - Send a newline character as message delimiter
 
-  - When delimiting newline (`"\n"`) character is received
-    - It parses the [message](#tcp-socket-message-format) in the buffer as JSON
-    - It looks for `event` key in received object and executes appropriate registered hooks functions
+## Termination
 
-    - When the hook function is being executed
-      - It passes value of `data` key from received object to the executed function
-      - Hook function is able to modify data
+When the testing is done, Dredd signals the hook handler process to terminate. This is done repeatedly with delays. When termination timeout is over, Dredd loses its patience and kills the process forcefully.
 
-    - When function was executed
-      - It should serialize message to JSON
-      - Send the serialized message back to the socket with same `uuid` as received
-      - Send a newline character as message delimiter
+- **retry delays** can be configured by [--hooks-worker-term-retry](usage-cli.md#-hooks-worker-term-retry)
+- **timeout** can be configured by [--hooks-worker-term-timeout](usage-cli.md#-hooks-worker-term-timeout)
+
+On Linux or macOS, Dredd uses the `SIGTERM` signal to tell the hook handler process it should terminate. On Windows, where signals do not exist, Dredd sends the `END OF TEXT` character (`\u0003`, which is ASCII representation of <kbd>Ctrl+C</kbd>) to standard input of the process.
```

Comment:
Unfortunately, MkDocs / ReadTheDocs render it just as code. <kbd>Boring!</kbd> Maybe in the future I'll attach my own CSS to render it as nicely as it's on GitHub.